### PR TITLE
Refactor tests to use vercel dev

### PR DESCRIPTION
**Pull request by de-luca**
Hi!

As evoked in #31, the tests seamed outdated.
I refactored them into using `vercel dev`.

This mainly remove the publication requirement.
However it seems like `vercel dev` cannot be used without linking to a project (see https://github.com/vercel/vercel/discussions/4925).

I kept what was beeing tested and only changed the way around.

This can also run on GitHub Actions, although it would require a vercel token.
(This can be seen here: https://github.com/de-luca/now-rust/actions?query=branch%3Agh)

### CODE REVIEW

File: test/fixtures/04-with-parameter/api/Cargo.toml.backup [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[package]
```
Comment (mike-engel) [inline test/fixtures/04-with-parameter/api/Cargo.toml.backup]: This file can probably be deleted, no? Or is there a reason a file with a `.backup` extension is included?

**Review by mike-engel — COMMENTED** [review]
@de-luca Thanks for this, and sorry for the delay. One comment, and could you rebase this with the latest master? Thanks!

File: test/fixtures/04-with-parameter/api/Cargo.toml.backup (hunk 1 above)
Comment (de-luca) [reply test/fixtures/04-with-parameter/api/Cargo.toml.backup]: Good catch it was leftover from my tests.

**Comment by de-luca** [comment]
I rebased the branch and cleaned the leftover (I hope I didn't took too long).
Tests are passing with #25 additions.

**Comment by mike-engel** [comment]
Hi @de-luca, no worries at all! I've been busy as well. It looks like there's one last remaining conflict in the `package-log.json` file. When you have time to take a look at that, I think this looks good to merge thereafter.

**Comment by mike-engel** [comment]
Thanks for this @de-luca and for your patience 😄

**Pull request merged**